### PR TITLE
fix(statuscolumn): fix sign column highlights

### LINKED ISSUE #21726: statuscolumn: sign highlight not work with cursorline
### Describe the bug

when set the cursorline and there has a sign . the sign  foreground color will be ignore.

### Steps to reproduce

```lua
vim.opt.stc = '%l%s%C'
vim.opt.cursorline = true

vim.cmd('sign define MySign text=>> texthl=Search')
vim.fn.sign_place(5, '', 'MySign', vim.api.nvim_buf_get_name(0), { lnum = 10 })
```

### Expected behavior

works like normal

### Neovim version (nvim -v)

NVIM v0.9.0-dev-648+g87cfe5094-dirty

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Macos 13.0.1

### Terminal name/version

kitty

### $TERM environment variable

xterm-kitty

### Installation

build frome source

### PR DESCRIPTION
**Pull request by luukvbaal**
Resolve https://github.com/neovim/neovim/issues/21726.

### CODE REVIEW

**Comment by luukvbaal** [comment]
Added a test to `sign_spec.lua` to verify that the highlights are now actually the same as with the default sign column.

**Comment by zeertzjq** [comment]
I think `sign_num_attr` is still needed. Consider the following test change:
```diff
diff --git a/test/functional/ui/sign_spec.lua b/test/functional/ui/sign_spec.lua
index a1683a32c..ae71707a8 100644
--- a/test/functional/ui/sign_spec.lua
+++ b/test/functional/ui/sign_spec.lua
@@ -156,6 +156,8 @@ describe('Signs', function()
         {0:~                                                    }|
                                                              |
       ]])
+      command('set statuscolumn=%s%l')
+      screen:expect_unchanged()
     end)
 
     it('highlights the cursorline sign with culhl', function()
```

File: src/nvim/drawline.c [hunk 1]
```diff
@@ -403,7 +403,7 @@ static int get_sign_attrs(buf_T *buf, linenr_T lnum, SignTextAttrs *sattrs, int
 ///
 /// @param[out] stcp  Status column attributes
 static void get_statuscol_str(win_T *wp, linenr_T lnum, int row, int startrow, int filler_lines,
-                              int cul_attr, int sign_num_attr, SignTextAttrs *sattrs,
```
Comment (zeertzjq) [inline src/nvim/drawline.c:406]: I think `sign_num_attr` is still needed. Consider the following test change:
```diff
diff --git a/test/functional/ui/sign_spec.lua b/test/functional/ui/sign_spec.lua
index a1683a32c..ae71707a8 100644
--- a/test/functional/ui/sign_spec.lua
+++ b/test/functional/ui/sign_spec.lua
@@ -156,6 +156,8 @@ describe('Signs', function()
         {0:~                                                    }|
                                                              |
       ]])
+      command('set statuscolumn=%s%l')
+      screen:expect_unchanged()
     end)
 
     it('highlights the cursorline sign with culhl', function()
```

Comment (luukvbaal) [reply src/nvim/drawline.c:406]: OK I'll check. Can you describe in words when `sign_num_attr` is supposed to be used lol? I thought it pertained to `set signcolumn=number`.

```c
  int sign_num_attr = 0;                // sign attribute for the number column
```

Comment (luukvbaal) [reply src/nvim/drawline.c:406]: But I see it is actually used in the `WL_NR` state which is why I guess I included it originally.

Comment (zeertzjq) [reply src/nvim/drawline.c:406]: It is used for `numhl` is signs.

Comment (luukvbaal) [reply src/nvim/drawline.c:406]: > It is used for `numhl` in signs.

Right, never realized while working on the code lol.

**Review by zeertzjq — APPROVED** [review]

**Comment by c0r73x** [comment]
It still wont respect `CursorLineSign`

**Pull request merged**